packages/protocol/contracts/L1/ProverPool.sol

```diff
@@ -373,6 +391,23 @@ contract ProverPool is EssentialContract, IProverPool {
         }
     }
 
+    // Calculates the user weight's when it stakes/unstakes/slashed
+    function _calcWeight2(
+        uint16 currentCapacity,
+        uint64 stakedAmount,
+        uint16 rewardPerGas
+    )
+        private
+        pure
+        returns (uint256)
+    {
+        if (currentCapacity == 0 || stakedAmount == 0 || rewardPerGas == 0) {
+            return 0;
+        } else {
+            return uint256(stakedAmount) / rewardPerGas / rewardPerGas;
```

Comment:
stakedAmount is uint32, rewardPerGas is also uint32, maybe we want to multiply stakedAmount with a const value (or `stakedAmount  << 64`) so the result is not likely to be 0?